vyxal/transpile.py

```diff
@@ -0,0 +1,293 @@
+        res += indent_str("ctx.context_values.pop()", indent)
+
+        return res
+    if isinstance(struct, structure.ForLoop):
+        # TODO (user/ysthakur) make it work with multiple variables
+        var = (
+            struct.names[0] if struct.names else f"LOOP{secrets.token_hex(16)}"
+        )
+        var = f"VAR_{var}"
+        return (
+            indent_str(f"for {var} in iterable(pop(stack, ctx=ctx)):", indent)
+            + indent_str(f"    ctx.context_values.append({var})", indent)
+            + transpile_ast(struct.body, indent + 1)
+            + indent_str("    ctx.context_values.pop()", indent)
+        )
+    if isinstance(struct, structure.WhileLoop):
+        return (
+            indent_str("condition = pop(stack, ctx=ctx)", indent)
+            + indent_str("while boolify(condition):", indent)
+            + indent_str("    ctx.context_values.append(condition)", indent)
+            + transpile_ast(struct.body, indent + 1)
+            + indent_str("    ctx.context_values.pop()", indent)
+            + transpile_ast(struct.condition, indent + 1)
+            + indent_str("    condition = pop(stack, ctx=ctx)", indent)
+        )
+    if isinstance(struct, structure.FunctionCall):
+        return f"stack += FN_{struct.name}(stack, ctx=ctx)"
+    if isinstance(struct, structure.FunctionDef):
+        parameter_total = 0
+        function_parameters = ""
+
+        for parameter_token in struct.parameters:
+            parameter = parameter_token.value
+            if parameter.isnumeric():
+                parameter_total += int(parameter)
+                function_parameters += (
+                    f"parameters += pop(stack, {int(parameter)}, ctx)\n"
+                )
+            elif parameter == "*":
+                function_parameters += (
+                    "parameters += " + "pop(stack, pop(stack, ctx=ctx), ctx)\n"
+                )
+            else:
+                parameter_total += 1
+                function_parameters += f"VAR_{parameter} = pop(stack, ctx=ctx)"
+
+            return (
+                indent_str(f"def FN_{struct.name}(stack, ctx)", indent)
+                + indent_str("stack = []", indent + 1)
+                + indent_str(f"this = FN_{struct.name}", indent + 1)
+                + indent_str("ctx.input_level += 1", indent + 1)
+                + indent_str(function_parameters, indent + 1)
+                + indent_str("stack = parameters", indent + 1)
+                + indent_str(
+                    "ctx.input_values[ctx.input_level] = [stack[::], 0]",
+                    indent + 1,
+                )
+                + transpile_ast(struct.body, indent + 1)
+                + indent_str("ctx.context_values.pop()", indent + 1)
+                + indent_str("ctx.input_level -= 1", indent + 1)
+                + indent_str("return stack", indent + 1)
+            )
+    if isinstance(struct, structure.Lambda):
+        signature = secrets.token_hex(16)
```

Comment:
I don't think `signature` is the right word. Signature typically refers to the arity and types of a function. Maybe just `id` would be better?

Reply:
`id` is a python built-in. But I agree that a name change would be suitable

Reply:
`id_` would be the standard workaround, but tbh, when have you ever used the `id` builtin? I normally just cope with it being shadowed